### PR TITLE
DiskOffering in CloudStackMachine CRD should be a pointer to CloudStackResourceDiskOffering

### PR DESCRIPTION
**Pull request by hrak**
*Issue #, if available:*

#326 

*Description of changes:*

This is a breaking change in the (unreleased) v1beta3 API which changes the `DiskOffering` property of `CloudStackMachine` to be a pointer to `CloudStackResourceDiskOffering`

Reasoning is described in the above linked issue.

*Testing performed:*

make test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by hrak** [comment]
While working on this PR i was surprised to find that API conversion code was not being generated anymore by the Makefile. I fixed that in this PR too + updated the webhooks to conform with requirements set in CAPI 0.4 (yes, 0.4)

**Review by vishesh92 — APPROVED** [review]
lgtm

File: Makefile [hunk 1]
```diff
@@ -158,12 +159,14 @@ config/.flag.mk: $(CONTROLLER_GEN) $(MANIFEST_GEN_INPUTS)
 	@touch config/.flag.mk
 
 CONVERSION_GEN_TARGET=$(shell find api -type d -name "v*1" -exec echo {}\/zz_generated.conversion.go \;)
-CONVERSION_GEN_INPUTS=$(shell find ./api -name "*test*" -prune -o -name "*zz_generated*" -prune -o -type f -print)
 .PHONY: generate-conversion
-generate-conversion: $(CONVERSION_GEN_TARGET) ## Generate code to convert api/v1beta1 to api/v1beta2
-api/%/zz_generated.conversion.go: bin/conversion-gen $(CONVERSION_GEN_INPUTS)
-	conversion-gen --go-header-file "./hack/boilerplate.go.txt" --input-dirs "./api/v1beta1" \
-	--output-base "." --output-file-base="zz_generated.conversion" --skip-unsafe=true
+generate-conversion: $(CONTROLLER_GEN) ## Generate code to convert api/v1beta1 and api/v1beta2 to api/v1beta3
```
Comment (vishesh92) [inline Makefile:163]:
```suggestion
generate-conversion: $(CONVERSION_GEN) ## Generate code to convert api/v1beta1 and api/v1beta2 to api/v1beta3
```

Comment (vishesh92) [reply Makefile:163]: @hrak **capi-provider-cloudstack-presubmit-e2e-smoke-test** is currently failing because it can't find `conversion-gen`.

Comment (rohityadavcloud) [inline Makefile:163]:
```suggestion
generate-conversion: $(CONVERSION_GEN) ## Generate code to convert api/v1beta1 and api/v1beta2 to api/v1beta3
```

**Comment by rohityadavcloud** [comment]
@vishesh92 pl fix the minor issue post-merging this PR, thanks.

**Review by vishesh92 — APPROVED** [review]

**Comment by vishesh92** [comment]
@hrak For some reason I am not able to resolve conflicts for this PR. If you have time, can you rebase this PR and resolve conflicts? Else I will have to create a duplicate PR.

**Comment by vishesh92** [comment]
I have recreated this PR here: https://github.com/kubernetes-sigs/cluster-api-provider-cloudstack/pull/359 
after fixing the conflicts and fixing the `make generate-conversion` command.

**Pull request closed without merging**